[GEOS-10444]: Supporting REPLACE UpdateMode on rasters import (#5781)
geoserver/geoserver: doc/en/user/source/extensions/importer/rest_reference.rst

A PUT request over an existing task can be used to update its representation. The representation can be partial, and just contains
the elements that need to be updated.

The updateMode of a task may have different values. 

.. list-table::
   :header-rows: 1

   * - UpdateMode
     - Description
   * - CREATE
     - This is the default starting updateMode of a task, that is: create the target resource if missing.
   * - REPLACE

>      - For vector stores: delete the existing features in the target layer and replace them with the task source ones. For raster stores: replace the underlying data. When dealing with StructuredGridCoverage reader (i.e. ImageMosaic) the new file will get harvested (replacing the old one). For single raster coverages (i.e. GeoTIFF) the name of the file should be the same so that the coverageStore layer will preserve the original name (The old file will be deleted).

Reviewer: ```suggestion
     - For vector stores: delete the existing features in the target layer and replace them with those from the task source. For raster stores: replace the underlying data. When dealing with StructuredGridCoverage reader (e.g. ImageMosaic) the new file will be harvested (replacing the old one). For single raster coverages (e.g. GeoTIFF) the name of the file should be the same so that the coverageStore layer will preserve the original name (the old file will be deleted).
```
